deltacat/aws/s3u.py

```diff
@@ -385,7 +398,7 @@ def download_manifest_entry(
     retrying = Retrying(
         wait=wait_random_exponential(multiplier=1, max=60),
         stop=stop_after_delay(30 * 60),
-        retry=retry_if_not_exception_type(NonRetryableError),
+        retry=retry_if_exception_type(RetryableDownloadTableError),
```

Comment:
There should not be any ambiguity here. We need to retry for any `RetryableError`

```suggestion
        retry=retry_if_exception_type(RetryableError),
```